package/gluon-mesh-vpn-wireguard/files/lib/netifd/proto/gluon_wireguard.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/sh
+# Copyright 2016-2017 Christof Schulze <christof@christofschulze.com>
+# Licensed to the public under the Apache License 2.0.
+
+. /lib/functions.sh
+. ../netifd-proto.sh
+init_proto "$@"
+
+proto_gluon_wireguard_init_config() {
+  no_device=1
+  available=1
+  renew_handler=1
+}
+
+proto_gluon_wireguard_renew() {
+  local config="$1"
+  echo wireguard RENEW: $@
+  ifdown $config
+  ifup $config
+}
+
+proto_gluon_wireguard_setup() {
+  local config="$1"
+  ifname=$(uci get network.$config.ifname) # we need uci here because nodevice=1 means the device is not part of the ubus structure
+
+  if [[ $(wg show all latest-handshakes |wc -l) -ge $(uci get wireguard.mesh_vpn_backbone.peer_limit) ]]; then
```

Comment:
Isn't `if [[ ... ]]` a bash only thing?

Reply:
I'm not sure but this looks correct to me as it is a comparison (-ge / greater equal).

Reply:
@mweinelt i just tested it, it also works with ash.
but: it is not necessary (tested, works with [] ) and we maybe shouldn't use it if it isn't.
only thing, maybe you have to use "" around the $() parts to be safe

Reply:
> @christf Have you added the time handling to the broker script?
no, is this enough of a problem to write to flash on every boot?

Reply:
I would like to leave this with [[ ]] - it is best practice in bash to use these tests and busybox supports these well.